cmd/minishift/cmd/start.go

```diff
@@ -409,26 +409,75 @@ func exportContainerImages(hostVm *host.Host, version string) {
 	fmt.Println(fmt.Sprintf("-- Exporting of OpenShift images is occuring in background process with pid %d.", exportCmd.Process.Pid))
 }
 
-// calculateDiskSizeInMB converts a human specified disk size like "1000MB" or "1GB" and converts it into Megabits
-func calculateDiskSizeInMB(humanReadableDiskSize string) int {
-	diskSize, err := units.FromHumanSize(humanReadableDiskSize)
+type sizeValidatorFunc func(name string, diskSize string) error
+type sizeConverterFunc func(input int64) int
+
+func calculateHumanSize(sizeValidator sizeValidatorFunc, name string, humanReadableSize string, unit string, sizeConverter sizeConverterFunc) (int, error) {
+	if stringUtils.HasOnlyNumbers(humanReadableSize) {
```

Comment:
Not sure what you mean with the defaults. Memory is 2GB and unit MiB. The same goes for disk size, 20GB and the unit is MB. Seems consistent... Changing this would cause the `config set` and flags `--memory` and `--disk-size`to act different from before